pkg/core/blockchain.go

```diff
@@ -408,13 +408,24 @@ func (bc *Blockchain) processHeader(h *block.Header, batch storage.Batch, header
 	return nil
 }
 
+// bc.GetHeaderHash(int(endHeight)) returns sum of all system fees for blocks up to h.
+// and 0 if no such block exists.
+func (bc *Blockchain) getSystemFeeAmount(h util.Uint256) uint32 {
+	_, sf, _ := bc.dao.GetBlock(h)
+	return sf
+}
+
 // TODO: storeBlock needs some more love, its implemented as in the original
 // project. This for the sake of development speed and understanding of what
 // is happening here, quite allot as you can see :). If things are wired together
 // and all tests are in place, we can make a more optimized and cleaner implementation.
 func (bc *Blockchain) storeBlock(block *block.Block) error {
 	cache := newCachedDao(bc.dao.store)
-	if err := cache.StoreAsBlock(block, 0); err != nil {
+	fee := bc.getSystemFeeAmount(block.PrevHash)
```

Comment:
We have `topBlock` in `Blockchain`, maybe it's worth saving sysFee the same way?

Reply:
`topBlock` is used to get top block/header without accessing the database.
With `sysFee` I see no situations where it can be useful. `sysFee` is needed in claim calculations and all blocks accessed there are surely not from the top.
With possible auto-claiming in NEO3.0 it can be useful though.